### ISSUE TITLE
Disable product that relate to another product make graphql break when query.

### ISSUE THREAD

**Issue by forfin** [magento/magento2#28892]
### Preconditions (*)

1. Magento Open Source 2.4-develop,  2.3.5-p1 fresh install

### Steps to reproduce (*)

1. Create first product (name: test-01).
2. Create another product (name: test-02). + Add test-01 as related products
3. Disable product test-01
4. Send graphql request to get relate product
```shell
curl -X POST \
  http://magento-2-fresh/graphql \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/json' \
  -d '{"query":"  query productDetail {\n    productDetail: products(pageSize: 10, currentPage: 1, search: \"\") {\n      items {\n        id\n        sku\n        name\n        related_products {\n          id\n          sku\n          url_key\n        }\n      }\n    }\n  }","operationName":"productDetail"}'
```
* Graphql format
```
query productDetail {
  productDetail: products(pageSize: 10, currentPage: 1, search: "") {
    items {
      id
      sku
      name
      related_products {
        id
        sku
        url_key
      }
    }
  }
}
```

### Expected result (*)

1. json response with no error
```json
{
    "data": {
        "productDetail": {
            "items": [
                {
                    "id": 2,
                    "sku": "Test-02",
                    "name": "Test-02",
                    "related_products": []
                }
            ]
        }
    }
}
```

### Actual result (*)

```json
{
    "errors": [
        {
            "debugMessage": "Notice: Undefined offset: 1 in /app/vendor/magento/module-related-product-graph-ql/Model/Resolver/Batch/AbstractLikedProducts.php on line 115",
            "message": "Internal server error",
            "extensions": {
                "category": "internal"
            },
            "locations": [
                {
                    "line": 7,
                    "column": 9
                }
            ],
            "path": [
                "productDetail",
                "items",
                0,
                "related_products"
            ]
        }
    ],
    "data": {
        "productDetail": {
            "items": [
                {
                    "id": 2,
                    "sku": "Test-02",
                    "name": "Test-02",
                    "related_products": null
                }
            ]
        }
    }
}
```

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [x] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [x] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by magento-engcom-team** [comment]
@engcom-Delta Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-37585` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @forfin. Thank you for your report. 
The issue has been fixed in magento/magento2#28896 by @forfin in 2.4-develop branch
Related commit(s):
 - [a57dbd6d203251a4dd93f5d3fb258d53d7464821](https://github.com/magento/magento2/commit/a57dbd6d203251a4dd93f5d3fb258d53d7464821)
 - [ed408534df19f764c5e7c8c385fed3da16cc5235](https://github.com/magento/magento2/commit/ed408534df19f764c5e7c8c385fed3da16cc5235)
 - [d8314a13c2d2cc71e78a58be7c907eef56078ae8](https://github.com/magento/magento2/commit/d8314a13c2d2cc71e78a58be7c907eef56078ae8)

The fix will be available with the upcoming 2.4.2 release.
